sources/osg/State.js

```diff
@@ -480,11 +490,18 @@ MACROUTILS.createPrototypeObject(
                         : stateSetTextureAttributeLength;
                 for (var j = 0; j < lt; j++) {
                     var attributeId = j;
-                    var attributeStack = _textureAttributeArray[attributeId];
-                    var stateSetAttributePair = stateSetTextureAttributeArray
-                        ? stateSetTextureAttributeArray[attributeId]
-                        : undefined;
-                    var hasStateAttributeStack = attributeStack !== undefined;
+
+                    var attributeStack =
+                        attributeId < _textureAttributeArray.length
```

Comment:
tested caching _textureAttributeArray.length and stateSetTextureAttributeArray.length as var outside of the loop ?

Reply:
fixed